src/index.js

```diff
@@ -3306,47 +3310,45 @@ async function retrieveMetadata() {
 
 	if ( queueItem ) {
 
-		let uri    = queueItem.dataset.file,
-			revoke = false;
+		let uri    = queueItem.dataset.file;
+		let file;
 
 		waitingMetadata++;
 		delete queueItem.dataset.retrieve;
+		let metadata;
 
-		queryMetadata: {
-			if ( queueItem.handle ) {
-				try {
-					if ( await queueItem.handle.requestPermission() != 'granted' )
-						break queryMetadata;
+		if ( queueItem.handle ) {
+			// Fetch metadata from File object
+			if ( await queueItem.handle.requestPermission() !== 'granted' )
+				return;
 
-					uri = URL.createObjectURL( await queueItem.handle.getFile() );
-					revoke = true;
-				}
-				catch( e ) {
-					break queryMetadata;
-				}
+			file = await queueItem.handle.getFile();
+			uri = URL.createObjectURL( file );
```

Comment:
We don't need this line anymore, as `uri` is now only used in the web fetch. We probably don't even need the variable at all.